pkgs/tools/backup/s3ql/default.nix

```diff
@@ -11,25 +11,45 @@ python3Packages.buildPythonApplication rec {
     sha256 = "1x0xj8clfs8fdczn8skc2wag5i4z47bsvlczn22iaf20hll1bb2w";
   };
 
-  checkInputs = [ which ] ++ (with python3Packages; [ cython pytest ]);
+  nativeBuildInputs = with python3Packages; [
+    cython
+  ];
+  buildInputs = [
+    sqlite
+  ];
+  checkInputs = with python3Packages; [
+    pytest
+  ] ++ [
+    which
+  ];
   propagatedBuildInputs = with python3Packages; [
-    sqlite apsw pycrypto requests defusedxml dugong llfuse
-    cython pytest pytest-catchlog google_auth google-auth-oauthlib
+    apsw
+    cryptography
+    defusedxml
+    dugong
+    google-auth-oauthlib
+    google_auth
+    llfuse
+    requests
+    systemd
+  ] ++ [
+    psmisc # `umount.s3ql` requires `fuser`
+    utillinuxMinimal.bin # `umount.s3ql` requires `umount`
```

Comment:
This will not put those dependencies on `PATH`. You will need to patch the paths in https://github.com/s3ql/s3ql/blob/02b4b5b0775e3e9fb61581d4ab917aa351802b43/src/s3ql/umount.py.

Reply:
I'm not sure exactly how s3ql uses `fuser` and `umount`. Looking at my logs from the time I made this patch, `umount.s3ql` gave errors like `FileNotFoundError: [Errno 2] No such file or directory: 'fuser': 'fuser'`. Adding those packages to `buildInputs` did not fix the errors, but adding them to `propagatedBuildInputs` did. Someone who understands the role of `propagatedBuildInputs` better than myself may have better insight.

Reply:
Please use makeWrapper instead of abusing propagatedBuildInputs.